azure-templates/build-op-steps.yml

```diff
@@ -0,0 +1,100 @@
+parameters:
+  - name: projectLocation
+    type: string
+  - name: buildOperation
+    type: string
+  - name: target
+    type: string
+    default: ''
+  - name: buildSpec
+    type: string
+    default: ''
+  - name: dependencies
+    type: object
+    default:
+      - file: ''
+
+steps:
+
+###########  Pull Dependencies in from latest build locations  ###########
+  - ${{ each dependency in parameters.dependencies }}: # TEMPORARY - default branchName is 'azure-pipelines' until things are actually committed to main on dependencies
+    - ${{ if ne( dependency.file, '' )}}:
+      - ${{ if eq( parameters.target, 'My Computer' )}}:
+        - powershell: Write-Output "##vso[task.setvariable variable=dependencyTargetDirectory]Windows"
+          displayName: set target for dependency to Windows
+      - ${{ if eq( parameters.target, 'Linux x64' )}}:
+        - powershell: Write-Output "##vso[task.setvariable variable=dependencyTargetDirectory]Linux_x64"
+          displayName: set target for dependency to Linux RT
+      - powershell: |
+          $branchName = 'azure-pipeline'
+          If (Test-Path -Path '${{ dependency.source }}\NI\export\$(Build.SourceBranchName)*\$(lvVersion)\$(architecture)') 
+          {
+            $branchName = '$(Build.SourceBranchName)'
+          }
+          $allDependencyBuilds = Get-ChildItem `
+            -Path "${{ dependency.source }}\NI\export\$branchName\*" `
+            | Sort {$_.Name} -Descending
+          $alreadyFound = $false
+          Foreach ($build in $allDependencyBuilds)
+          {
+            If(((Test-Path -Path "$build\$(lvVersion)\$(architecture)\.finished") -and -not($alreadyFound)))
+            {
+              $alreadyFound = $true
```

Comment:
Can we just use `break` instead of tracking with `alreadyFound`?

Reply:
break used in commit 6